api/src/main/java/org/apache/iceberg/RewriteManifests.java

```diff
@@ -49,9 +51,35 @@
    * that do not match the predicate are kept as-is. If this is not called and no predicate is set, then
    * all manifests will be rewritten.
    *
+   * Note that either clusterBy/rewriteIf should be used or deleteManifest/addManifest, not both.
+   *
    * @param predicate Predicate used to determine which manifests to rewrite. If true then the manifest
    *                  file will be included for rewrite. If false then then manifest is kept as-is.
    * @return this for method chaining
    */
   RewriteManifests rewriteIf(Predicate<ManifestFile> predicate);
+
+  /**
+   * Deletes a {@link ManifestFile manifest file} from the table. This method should be used
+   * together with {@link #addManifest(ManifestFile)} when manifests are rewritten using
+   * an external process.
```

Comment:
This should also note that manifest rewrites must produce the same set of data files between the deleted manifests and the added manifests.

Reply:
Added to the Javadoc.